public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -102,6 +102,38 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
     return getBackendSrv().fetch<Record<string, any>>(req);
   }
 
+  getLogsVolumeQuery(request: DataQueryRequest<LokiQuery>): Observable<DataQueryResponse> | undefined {
+    const isLogsVolumeAvailable = request.targets.some((target) => !isMetricsQuery(target.expr) && !!target.expr);
+    if (!isLogsVolumeAvailable) {
+      return;
```

Comment:
```suggestion
      return undefined;
```

this is just a personal opinion, feel free to ignore it: i recommend writing out the return-undefined explicitly in this case, because we do return `Observable|undefined`.

Reply:
Fixed in #39042 (instead of undefined, an empty collection of data providers is returned)